packages/runner/src/traverse.ts

```diff
@@ -625,6 +626,52 @@ export class SchemaObjectTraverser<K, S> extends BaseObjectTraverser<K, S> {
     }
   }
 
+  static resolveSchemaRef(
+    schemaContext: Readonly<SchemaContext>,
+  ): Readonly<SchemaContext> | undefined {
+    if (
+      !isObject(schemaContext.schema) || !("$ref" in schemaContext.schema) ||
+      schemaContext.schema["$ref"] === undefined
+    ) {
+      // We shouldn't have called this function, but just return it as-is
+      return schemaContext;
+    }
+    if (schemaContext.rootSchema === undefined) {
```

Comment:
maybe to phase rootSchema out more slowly, just use the schema instead if there is no rootSchema?

Reply:
except in the '#' case below, which isn't valid without a rootSchema